### PR TITLE
feat: provide param to control whether to build FTS index with positions

### PR DESCRIPTION
**Pull request by BubbleCal**
The positions can be very large, it's a waste for users who don't need phrase query, so provide a param to build the index without positions, don't build FTS index with positions by default

### CODE REVIEW

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -1280,6 +1281,15 @@ def create_scalar_index(
...
             Replace the existing index if it exists.
 
+        Optional Parameters
+        -------------------
+        with_position: bool, default True
+            This is for the ``INVERTED`` index. If True, the index will store the
+            positions of the words in the document, so that you can conduct phrase
+            query. This will significantly increase the index size.
+            It won't impact the performance of non-phrase queries even if it is set to
+            True.
```
Comment (wjones127) [inline python/python/lance/dataset.py:1291]: Do mean `False` here?
```suggestion
            It won't impact the performance of non-phrase queries even if it is set to
            False.
```

**Review by wjones127 — COMMENTED** [review]
All looks reasonable, just one question on one of the docs.

File: python/python/lance/dataset.py (hunk 1 above)
Comment (BubbleCal) [reply python/python/lance/dataset.py:1291]: It's `True`, we won't load the positions for non-phrase query, so it won't impact the non-phrase query performance

Comment (wjones127) [reply python/python/lance/dataset.py:1291]: Oh I see.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**